Example/Tests/AlbumsPickerViewControllerTests.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  AlbumsPickerViewControllerTests.swift
+//  GiniVision_Tests
+//
+//  Created by Enrique del Pozo Gómez on 2/26/18.
+//  Copyright © 2018 Gini GmbH. All rights reserved.
+//
+
+import XCTest
+@testable import GiniVision
+
+final class AlbumsPickerViewControllerTests: XCTestCase {
+    
+    let galleryManager = GalleryManagerMock()
+    lazy var vc = AlbumsPickerViewController(galleryManager: self.galleryManager)
+    
+    override func setUp() {
+        super.setUp()
+        _ = vc.view
+    }
+    
+    func testViewControllerTitle() {
+        let title = NSLocalizedStringPreferred("ginivision.albums.title",
+                                               comment: "title for the albums picker view controller")
+        XCTAssertEqual(title, vc.title, "title should match the one provided in the Localizable file")
+    }
+    
+    func testNumberOfSections() {
+        XCTAssertEqual(vc.albumsTableView.numberOfSections, 1, "There should be only one section")
+    }
+    
+    func testNumberOfItems() {
+        XCTAssertEqual(vc.albumsTableView.numberOfRows(inSection: 0), 3, "There should be 4 albums")
+    }
+    
+    func testCollectionCellType() {
+        XCTAssertNotNil(vc.tableView(vc.albumsTableView,
+                                     cellForRowAt: IndexPath(row:0, section:0)) as? AlbumsPickerTableViewCell,
+                        "cell type should match UITableViewCell")
+    }
+    
+    func testTableCellHeight() {
+        XCTAssertEqual(vc.tableView(vc.albumsTableView, heightForRowAt: IndexPath(row: 1, section:0)),
+                       AlbumsPickerTableViewCell.height,
+                       "table view cell heght should match AlbumsPickerTableViewCell height")
+    }
+    
+    func testTableCellSelection() {
+        let delegate = AlbumsPickerViewControllerDelegateMock()
+        vc.delegate = delegate
+        
+        let selectedIndex = IndexPath(row: 0, section: 0)
+        let selectedAlbum = galleryManager.albums[selectedIndex.row]
+        vc.tableView(vc.albumsTableView, didSelectRowAt: selectedIndex)
```

Comment:
Wouldn't it be better if you just selected the cell.
I guess you don't have to test that the `didSelectRowAt ` is called, but it's possible that the table view has selection disabled or something. It might be a better way to test it.

Reply:
`didSelectRowAt` method is not triggered when you select a cell programmatically unfortunately. UIKit rules.

Reply:
Oh... yeah, now I remember. I've had the same problem. It's intended behaviour.
It actually makes sense to a degree. For most use cases (that are not testing), if you select a cell in code, you wouldn't want to trigger the action for it.